Mathlib/Data/Set/Basic.lean

```diff
@@ -1521,6 +1521,36 @@ theorem eq_empty_of_ssubset_singleton {s : Set α} {x : α} (hs : s ⊂ {x}) : s
   ssubset_singleton_iff.1 hs
 #align set.eq_empty_of_ssubset_singleton Set.eq_empty_of_ssubset_singleton
 
+theorem Nonempty.eq_singleton_of_nonempty [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {default} := h.subset_singleton_iff.mp (fun x _ => by simp [Unique.uniq _ x])
+
+theorem Nonempty.eq_singleton_zero_of_nonempty [Zero α] [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {0} := by convert h.eq_singleton_of_nonempty
+
+theorem Nonempty.eq_singleton_one_of_nonempty [One α] [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {1} := by convert h.eq_singleton_of_nonempty
+
+noncomputable def uniqueOfNonempty [Subsingleton α] {s : Set α} (h : s.Nonempty) :
+    Unique α := uniqueOfSubsingleton h.choose
+
+theorem Nonempty.eq_singleton_of_nonempty_of_subsingleton [Subsingleton α] {s : Set α}
+    (h : s.Nonempty) :
+    letI := uniqueOfNonempty h
+    s = {default} := by
+  letI := uniqueOfNonempty h
+  exact h.eq_singleton_of_nonempty
+
+theorem subset_eq_of_subsingleton [Subsingleton α] {s : Set α} (h : s.Nonempty) (a : α) :
+    s = {a} := by rw [h.eq_singleton_of_nonempty_of_subsingleton, Unique.uniq _ a]
+
+@[simp]
```

Comment:
i worry about these lemmas being `@[simp]` but I think that this is not what most people think, I just think they'll be slow

Reply:
No, you are right. These should not be [simp]. I forgot to change it back.